craft_parts/utils/partition_utils.py

```diff
@@ -118,58 +115,123 @@ def _validate_partition_naming_convention(partitions: Sequence[str]) -> None:
                 ),
             )
 
+        if _is_valid_partition_name(partition):
+            continue
         raise errors.FeatureError(
             message=f"Partition {partition!r} is invalid.",
             details=PARTITION_INVALID_MSG,
         )
 
 
-def _validate_namespace_conflicts(partitions: Sequence[str]) -> None:
-    """Validate conflicts between regular partitions and namespaces.
+def _validate_partitions_conflicts(partitions: Sequence[str]) -> None:
+    """Validate conflicts between partitions.
+
+    Assumes partition names are valid and unique.
+
+    :raises FeatureError: for conflicts
+    """
+    conflictings_partitions = detect_conflicts(partitions)
+
+    if not conflictings_partitions:
+        return
+
+    filtered_conflictings_partitions = unique_conflicts(conflictings_partitions)
+
+    # Raise the full list of collisions
+    lines: list[str] = []
+    for conflictings in filtered_conflictings_partitions.values():
+        conflictings_list = list(conflictings)
+        conflictings_list.sort()
+        lines.append(f"- {str(conflictings_list)[1:-1]}")
+
+    msg = "\n".join(lines)
+
+    raise errors.FeatureError(
+        message=f"Following partitions conflicts:\n{msg}",
+        details="Hyphens and slashes are converted to underscores to associate partitions names with environment variables. 'foo-bar' and 'foo/bar' would result in environment variable FOO_BAR.",
```

Comment:
I think we're using the name "dash" for - elsewhere (@mr-cal, can you confirm?) 
Also change the message to something like "There are partition name conflicts"?

Reply:
I think "hyphen" is used in most places across our repos.